src/api/app/controllers/webui/staging_workflows_controller.rb

```diff
@@ -1,25 +1,44 @@
 class Webui::StagingWorkflowsController < Webui::WebuiController
   layout 'webui2/webui'
 
+  before_action :require_login, except: [:show]
   before_action :set_bootstrap_views
-
   before_action :set_project, only: [:new, :create]
+  after_action :verify_authorized, except: [:show, :new]
 
-  def new; end
+  def new
+    @staging_workflow = StagingWorkflow.new
+    redirect_to staging_workflow_path(@project.staging) if @project.staging
+  end
 
   def create
-    staging_workflow = @project.staging.new
-    if staging_workflow.save
-      flash[:success] = "Staging Workflow for #{@project.name} was successfully created"
-      redirect_to :show
-    else
-      flash[:error] = "Staging Workflow for #{@project.name} couldn't be created"
-      redirect_to :new
+    staging_workflow = @project.build_staging
+
+    authorize staging_workflow
+
+    StagingWorkflow.transaction do
+      if staging_workflow.save
+        ['A', 'B'].each do |letter|
```

Comment:
The number of projects should be responsibility of the StagingWorkflow class. 

E.g.
https://stackoverflow.com/a/18233271

Reply:
Will do, it feels better then :+1: